core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -61,6 +66,22 @@ public WebSocketService(SonarLintClient client, ConnectionConfigurationRepositor
       .dispatch(QualityGateChangedEvent.class, new ShowSmartNotificationOnQualityGateChangedEvent(client, configurationRepository));
   }
 
+  @PostConstruct
+  public void initialize() {
+    ScheduledExecutorService webSocketConnectionRefresher = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "WebSocket connection refresher"));
+    webSocketConnectionRefresher.scheduleAtFixedRate(this::refreshConnectionIfNeeded, 119, 119, TimeUnit.MINUTES);
```

Comment:
Raising another option as it was not described in the PR summary: we could schedule the task in the `createConnectionIfNeeded` method, so we know the task runs exactly 119min after a connection was opened. But it means you might need to cancel the previous task. So a bit more complicated than this implementation